### DISCUSSION TITLE
Message: x-hasura-admin-secret but not found,

### DISCUSSION THREAD

**Discussion by mvltyldrmm** [shurcooL/graphql#96] [Q&A] [answered]
**Hi. 
My graphql is live and on hasura. I need the key for connection. But I don't understand how to get it for this library. How can I run queries and mutations using secret key?**

**Comment by celikezer** [comment, marked as answer]
I have had the same issue, and I solve it by passing these missing parameters to the header like below.

Create a round trip with your required tokens & credentials or etc. Then create an HTTP client using this Round Trip, this will add the header parameters, and you can edit your requests there.

> You don't need the use oauth2 or other clients, the standard Go HTTP client is enough since Hasura API does not accept Bearer token I guess because even if you provide this token, it still asks for missing header parameters.

```go
var client *graphql.Client

func init() {
	if client != nil {
		return
	}

	httpClient := http.Client{
		Transport: &authedTransport{
			wrapped: http.DefaultTransport,
		},
	}

	client = graphql.NewClient("https://your-hasura-address.hasura.app/v1/graphql", &httpClient)
}

func GetGraphQLClient() *graphql.Client {
	return client
}

type authedTransport struct {
	wrapped http.RoundTripper
}

func (t *authedTransport) RoundTrip(req *http.Request) (*http.Response, error) {
        // you can mutate your requests here as you wish

	req.Header.Set("x-hasura-admin-secret", "your-admin-secret-key")
	req.Header.Set("x-hasura-access-key", "your-token")
	return t.wrapped.RoundTrip(req)
}
```
